### PR TITLE
docs(docs-site): add definitions and explanations for bonds

### PR DESCRIPTION
**Pull request by RogerLamTd**

### CODE REVIEW

File: packages/docs-site/src/content/docs/resources/terminology.mdx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+title: Taiko-specific Terminology
```
Comment (d1onys1us) [inline packages/docs-site/src/content/docs/resources/terminology.mdx:2]: prefer this to match the filename in kebab case. so can update this to terminology, or the filename to taiko-specific-terminology. also sentence case.

File: packages/docs-site/src/content/docs/resources/terminology.mdx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+- **Liveness Bond**: The bond provided by a block's assigned prover, serving as a commitment to initially prove the first transition and to provide post-contest proofs within the corresponding proving window.
+
+- **Validity Bond**: The bond provided by the actual prover of a proof, signifying their commitment that the proof is indeed correct and that they are ready to withstand a contest.
+
+- **Contest Bond**: The bond provided by the contester.
+
+In layman's terms, the above terms relate to the protocol in the following manner:
+
+- A **Liveness Bond** is deposited when a prover gets assigned a block to prove; when the block gets verified it gets returned to the prover.
```
Comment (d1onys1us) [inline packages/docs-site/src/content/docs/resources/terminology.mdx:18]: consistency with bulleted lists. will this render actually? need a line break i think.

**Review by dionysuzx — APPROVED** [review]

**Review by davidtaikocha — APPROVED** [review]
